File: AndroidBasics/News App/app/src/main/java/com/mehru/newsapp/webView.java
package com.mehru.newsapp;

import android.content.Intent;
import android.os.Bundle;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Toolbar;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class webView extends AppCompatActivity {

    Toolbar toolbar;
    WebView webView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_web_view);


        toolbar = findViewById(R.id.toolbar);
        webView= findViewById(R.id.webView);
        setSupportActionBar(toolbar);
        Intent intent = getIntent();
        String url = "Url";
        webView.setWebViewClient(new WebViewClient());
        webView.loadUrl(url);
    }

    private void setSupportActionBar(Toolbar toolbar) {
    }
}

File: app dev kt/practice/app/src/main/java/com/mehru/practice/webView.kt
package com.mehru.practice

import android.os.Build
import android.os.Bundle
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.activity.enableEdgeToEdge
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.time.temporal.WeekFields

class webView : AppCompatActivity() {
    @RequiresApi(Build.VERSION_CODES.O)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_web_view)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val web = findViewById<WebView>(R.id.web)
        setweb(web)

    }


    @RequiresApi(Build.VERSION_CODES.O)
    private fun setweb(mehru: WebView) {
        mehru.webViewClient = WebViewClient()
        mehru.apply {
            settings.javaScriptEnabled = true
            settings.safeBrowsingEnabled = true
            loadUrl("https://github.com/Mehruuban")
        }
    }
}